src/current/v24.3/triggers.md

```diff
@@ -0,0 +1,505 @@
+---
+title: Triggers
+summary: A trigger executes a function when one or more specified SQL operations is performed on a table.
+toc: true
+---
+
+{{site.data.alerts.callout_info}}
+{% include feature-phases/preview.md %}
+{{site.data.alerts.end}}
+
+A *trigger* executes a function when one or more specified SQL operations is performed on a table. The executed function is called a [*trigger function*](#trigger-function) and is written in [PL/pgSQL]({% link {{ page.version.version }}/plpgsql.md %}).
+
+Triggers respond to data changes by adding logic within the database, rather than in an application. They can be used to modify data before it is inserted, maintain data consistency across rows or tables, or record an update to a row.
+
+## Structure
+
+A trigger consists of a trigger name, table name associated with the trigger, SQL operations and other conditions that activate the trigger, and a trigger function name with optional arguments. A trigger is defined with [`CREATE TRIGGER`]({% link {{ page.version.version }}/create-trigger.md %}) and has the following overall structure:
+
+~~~ sql
+CREATE TRIGGER trigger_name 
+  [ BEFORE | AFTER ] [ INSERT | UPDATE | DELETE ] ON table_name
+  FOR EACH ROW
+  [ WHEN boolean_condition ]
+  EXECUTE FUNCTION function_name(arguments)
+~~~
+
+- The trigger can activate `BEFORE` or `AFTER` any combination of `INSERT`, `UPDATE`, or `DELETE` statements is issued on a given table.
+	- `FOR EACH ROW` specifies a row-level trigger, which activates once for each row that is affected by the statements.
+	- `WHEN` specifies an optional boolean condition that determines whether the trigger activates for a given row.
+	- For details on the preceding behaviors, refer to [Trigger conditions](#trigger-conditions).
+- The [trigger function](#trigger-function), written in [PL/pgSQL]({% link {{ page.version.version }}/plpgsql.md %}), is executed each time the trigger activates. A comma-separated list of function arguments can be included.
+
+### Trigger conditions
+
+A trigger activates when one or more SQL statements is issued on a table. The statement can be [`INSERT`]({% link {{ page.version.version }}/insert.md %}), [`DELETE`]({% link {{ page.version.version }}/delete.md %}), or [`UPDATE`]({% link {{ page.version.version }}/update.md %}). 
+
+To specify more than one statement, use the `OR` clause. For example:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TRIGGER check_value
+  BEFORE INSERT OR UPDATE ON users
+  ...
+~~~
+
+{{site.data.alerts.callout_info}}
+Triggers on `INSERT` and `UPDATE` also activate when [`UPSERT`]({% link {{ page.version.version }}/upsert.md %}) statements insert or update rows, respectively. However, `UPSERT` cannot be specified in a [`CREATE TRIGGER`]({% link {{ page.version.version }}/create-trigger.md %}) statement.
+{{site.data.alerts.end}}
```

Comment:
Another thing to call out is that `UPDATE` triggers can fire if the `ON CONFLICT` clause of an `INSERT` updates rows.